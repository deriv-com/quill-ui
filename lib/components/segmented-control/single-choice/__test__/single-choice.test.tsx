import { render } from "@testing-library/react";
import SingleChoice from "..";

describe("SingleChoice", () => {
    it("Should render SingleChoice Group With Icons Only", () => {
        const { container } = render(<SingleChoice.GroupWithIconsOnly />);
        expect(container).toMatchSnapshot();
    });
    it("Should render SingleChoice Group With Labels Only", () => {
        const { container } = render(<SingleChoice.GroupWithLabelsOnly />);
        expect(container).toMatchSnapshot();
    });
    it("Should render SingleChoice Group With Icons and Labels", () => {
        const { container } = render(<SingleChoice.GroupWithIconsAndLabels />);
        expect(container).toMatchSnapshot();
    });
});
