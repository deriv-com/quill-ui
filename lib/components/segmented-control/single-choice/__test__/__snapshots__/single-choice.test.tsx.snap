// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControlSingleChoice should inherit container width 1`] = `
<div>
  <div
    style="width: 328px;"
  >
    <div
      class="segmented-control segmented-control--md segmented-control--has-container-width"
    >
      <div
        class="item selected"
      >
        <span
          class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
        >
          Label
        </span>
      </div>
      <div
        class="item"
      >
        <span
          class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
        >
          Label
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SegmentedControlSingleChoice should render segments with icons and labels 1`] = `
<div>
  <div
    class="segmented-control segmented-control--md"
  >
    <div
      class="item selected"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
  </div>
</div>
`;

exports[`SegmentedControlSingleChoice should render segments with icons only 1`] = `
<div>
  <div
    class="segmented-control segmented-control--md"
  >
    <div
      class="item selected"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <div>
          IconComponent
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`SegmentedControlSingleChoice should render segments with labels only 1`] = `
<div>
  <div
    class="segmented-control segmented-control--md"
  >
    <div
      class="item selected"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
  </div>
</div>
`;

exports[`SegmentedControlSingleChoice should render segments with placeholder icons 1`] = `
<div>
  <div
    class="segmented-control segmented-control--md"
  >
    <div
      class="item selected"
    >
      <span
        class="icon"
      >
        <svg
          height="24"
          role="img"
          viewBox="0 0 14 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
            />
          </g>
          <defs>
            <clippath
              id="53a89012ab3a8648185823a89f2a6ff9__a"
            >
              <path
                d="M0 0h14v24H0z"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <svg
          height="24"
          role="img"
          viewBox="0 0 14 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
            />
          </g>
          <defs>
            <clippath
              id="53a89012ab3a8648185823a89f2a6ff9__a"
            >
              <path
                d="M0 0h14v24H0z"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <svg
          height="24"
          role="img"
          viewBox="0 0 14 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
            />
          </g>
          <defs>
            <clippath
              id="53a89012ab3a8648185823a89f2a6ff9__a"
            >
              <path
                d="M0 0h14v24H0z"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <svg
          height="24"
          role="img"
          viewBox="0 0 14 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
            />
          </g>
          <defs>
            <clippath
              id="53a89012ab3a8648185823a89f2a6ff9__a"
            >
              <path
                d="M0 0h14v24H0z"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="icon"
      >
        <svg
          height="24"
          role="img"
          viewBox="0 0 14 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
            />
          </g>
          <defs>
            <clippath
              id="53a89012ab3a8648185823a89f2a6ff9__a"
            >
              <path
                d="M0 0h14v24H0z"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`SegmentedControlSingleChoice should render with one disabled option 1`] = `
<div>
  <div
    class="segmented-control segmented-control--md"
  >
    <div
      class="item selected"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item disabled"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
    <div
      class="item"
    >
      <span
        class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Label
      </span>
    </div>
  </div>
</div>
`;
